feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/presentation/view/AlertsView.kt

```diff
@@ -0,0 +1,77 @@
+package jp.co.soramitsu.feature_staking_impl.presentation.view
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.View
+import android.widget.LinearLayout
+import androidx.annotation.StringRes
+import androidx.recyclerview.widget.LinearLayoutManager
+import jp.co.soramitsu.common.utils.makeGone
+import jp.co.soramitsu.common.utils.makeVisible
+import jp.co.soramitsu.common.view.shape.getCutCornerDrawable
+import jp.co.soramitsu.feature_staking_impl.R
+import jp.co.soramitsu.feature_staking_impl.presentation.staking.alerts.model.AlertModel
+import jp.co.soramitsu.feature_staking_impl.presentation.staking.alerts.AlertsAdapter
+import kotlinx.android.synthetic.main.view_alert.view.*
+
+class AlertsView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0,
+) : LinearLayout(context, attrs, defStyle), AlertsAdapter.ItemHandler {
+
+    sealed class Status(@StringRes val textRes: Int?, @StringRes val extraTextRes: Int?) {
+        class Alerts(val alerts: List<AlertModel>) : Status(textRes = null, extraTextRes = null)
```

Comment:
I don't like those null values. May you suggest nicer way? @valentunn

Reply:
I removed two wrong states for the widget. So we don't need variables in constructor now, and so there is no that problem